src/array.cr

```diff
@@ -2021,16 +2024,28 @@ class Array(T)
       # `[-, -, -, 'c', 'd', -]`
       shift_buffer_by(half_capacity)
     else
-      double_capacity_for_unshift
+      increase_capacity_for_unshift
     end
   end
 
   def remaining_capacity : Int32
     @capacity - @offset_to_buffer
   end
 
-  private def double_capacity
-    resize_to_capacity(@capacity == 0 ? 3 : (@capacity * 2))
+  private def calculate_new_capacity
+    if @capacity == 0
+      3
```

Comment:
I wonder where this 3 came from in the original code. Wouldn't it be better to be a power of two (e.g. 4)?

Reply:
@asterite ?

Reply:
Lucky number 😄